Source/items.cpp

```diff
@@ -855,7 +855,7 @@ int SaveItemPower(const Player &player, Item &item, ItemPower &power)
 		RedrawComponent(PanelDrawComponent::Mana);
 		break;
 	case IPL_DUR: {
-		int bonus = r * item._iMaxDur / 100;
+		int bonus = std::min(r * item._iMaxDur / 100, DUR_INDESTRUCTIBLE - item._iMaxDur);

```

Comment:
I don't think we want to make them indestructible, so this PR either depends on https://github.com/diasurgical/devilutionX/pull/5772 or the following change should be made:

```suggestion
		int bonus = std::min(r * item._iMaxDur / 100, DUR_INDESTRUCTIBLE - item._iMaxDur - 1);
```